src/main/java/tasks/TaskList.java

```diff
@@ -0,0 +1,160 @@
+     *
+     * @param str The input command
+     * @return A response from the bot to indicate a deadline task has been created
+     * and added to the task list
+     */
+    public String deadlineTask(String str) {
+        try {
+            int slashIndex = str.indexOf("/");
+            String day = str.substring(slashIndex + 4, slashIndex + 14);
+            String time = str.substring(slashIndex + 14);
+            Task task = new Deadline(str.substring(0, slashIndex), Storage.formatDate(day) + time);
+            tasks.add(task);
+            Storage.writeLine(task);
+            return "Got it. I've added this task: \n"
+                    + task
+                    + "\nNow you have " + tasks.size() + " tasks in the list.";
+        } catch (StringIndexOutOfBoundsException e) {
+            return "☹ OOPS!!! The description of a deadline cannot be empty.\n" + 
+                    "If you have entered a day instead of a date, please enter a date in the format: yyyy-mm-dd.";
+        } catch (DateTimeParseException e) {
+            return "☹ OOPS!!! Please use the date format: yyyy-mm-dd.";
+        }
+    }
+
+    /**
+     * A method to create & add an event
+     *
+     * @param str The input command
+     * @return A response from the bot to indicate a event has been created
+     * and added to the task list
+     */
+    public String eventsTask(String str) {
+        try {
+            int slashIndex = str.indexOf("/");
+            String day = str.substring(slashIndex + 4, slashIndex + 14);
+            Task task = new Events(str.substring(0, slashIndex), day);
+            tasks.add(task);
+            Storage.writeLine(task);
+            return "Got it. I've added this task: \n"
+                    + task
+                    + "\nNow you have " + tasks.size() + " tasks in the list.";
+        } catch (StringIndexOutOfBoundsException e) {
+            return "☹ OOPS!!! The description of a deadline cannot be empty.\n" +
+                    "If you have entered a day instead of a date, please enter a date in the format: yyyy-mm-dd.";
+        } catch (DateTimeParseException e) {
+            return "☹ OOPS!!! Please use the date format: yyyy-mm-dd.";
+        }
+    }
+
+    /**
+     * A method to delete a task
+     *
+     * @param str The input command
+     * @return A response from the bot to indicate a task has been deleted
+     * and added to the task list
+     */
+    public String deleteTask(String str) {
+        int indexOfTaskToDelete = Integer.parseInt(str.substring(7)) - 1;
+        Task t = tasks.get(indexOfTaskToDelete);
+        tasks.remove(indexOfTaskToDelete);
+        Storage.getAllTasks(tasks);
+        return "Got it. I've deleted this task: \n"
+                + t
+                + "\nNow you have " + tasks.size() + " tasks in the list.";
+    }
+
+    /**
+     * Command to find and display all tasks containing keyword 
+     * @param keyword word to search for
```

Comment:
Should there be a blank line after the first statement and before the param in the comment?